RegisterPayments/handler.ts

```diff
@@ -0,0 +1,325 @@
+              .mapNullable(getElementTextContent)
+              .map(importoTotalePagato => ({
+                ..._,
+                data: { ..._.data, importoTotalePagato }
+              }))
+          )
+        )
+        .chain(_ =>
+          fromOption(new Error("Missing datiSingoloPagamento"))(
+            fromNullable(
+              _.elements.datiPagamento
+                .getElementsByTagName("datiSingoloPagamento")
+                .item(0)
+            ).chain(datiSingoloPagamento =>
+              fromNullable(
+                datiSingoloPagamento
+                  .getElementsByTagName("singoloImportoPagato")
+                  .item(0)
+              )
+                .mapNullable(getElementTextContent)
+                .map(singoloImportoPagato => ({
+                  data: {
+                    ..._.data,
+                    singoloImportoPagato
+                  },
+                  elements: {
+                    ..._.elements,
+                    datiSingoloPagamento
+                  }
+                }))
+            )
+          )
+            .chain(_1 =>
+              fromOption(new Error("Missing dataEsitoSingoloPagamento"))(
+                fromNullable(
+                  _1.elements.datiSingoloPagamento
+                    .getElementsByTagName("dataEsitoSingoloPagamento")
+                    .item(0)
+                )
+                  .mapNullable(getElementTextContent)
+                  .map(dataEsitoSingoloPagamento => ({
+                    data: {
+                      ..._1.data,
+                      dataEsitoSingoloPagamento
+                    },
+                    elements: _1.elements
+                  }))
+              )
+            )
+            .chain(_1 =>
+              fromOption(new Error("Missing identificativoUnivocoRiscossione"))(
+                fromNullable(
+                  _1.elements.datiSingoloPagamento
+                    .getElementsByTagName("identificativoUnivocoRiscossione")
+                    .item(0)
+                )
+                  .mapNullable(getElementTextContent)
+                  .map(identificativoUnivocoRiscossione => ({
+                    data: {
+                      ..._1.data,
+                      identificativoUnivocoRiscossione
+                    },
+                    elements: _1.elements
+                  }))
+              )
+            )
+            .chain(_1 =>
+              fromOption(new Error("Missing causaleVersamento"))(
+                fromNullable(
+                  _1.elements.datiSingoloPagamento
+                    .getElementsByTagName("causaleVersamento")
+                    .item(0)
+                )
+                  .mapNullable(getElementTextContent)
+                  .map(causaleVersamento => ({
+                    data: { ..._1.data, causaleVersamento },
+                    elements: _1.elements
+                  }))
+              )
+            )
+            .chain(_1 =>
```

Comment:
another way to structure this code is to have some "small task" (one for each call to `chain` here, ie. getCausaleVersamento) defined outside this method, then compose them using sequence / sequenceS / sequenceT (let's suppose, IOEithers if they are sync or TaskEithers in case they are async).

Reply:
fixed in 07419b9